src/common/commonutils/CommandUtils.c

```diff
@@ -29,270 +31,194 @@ static int NormalizeStatus(int status)
-            waitpid(0, &intermediateStatus, 0);
-            _exit(status);
-        }
-        else if (intermediateProcess > 0)
-        {
-            // Wait on the intermediate process to finish
-            waitpid(intermediateProcess, &status, 0);
-        }
-        else
-        {
-            status = -1;
-            OsConfigLogError(log, "Failed forking intermediate process (errno: %d, '%s')", errno, strerror(errno));
-        }
-    }
-    else
-    {
-        OsConfigLogDebug(log, "SystemCommand: executing command '%s' without timeout or cancelation on %s thread", command, mainProcessThread ? "main process" : "worker");
-        if (0 == (workerProcess = fork()))
-        {
-            // Worker process
-            status = execl("/bin/sh", "sh", "-c", command, (char*)NULL);
-            _exit(status);
-        }
-        else if (workerProcess > 0)
-        {
-            // Wait on the worker process to terminate
-            waitpid(workerProcess, &status, 0);
-        }
-        else
-        {
-            // If our fork fails, try system(), if that also fails then the call fails
-            OsConfigLogError(log, "Failed forking process to execute command (errno: %d, '%s'), attempting system", errno, strerror(errno));
-            status = system(command);
-        }
-    }
-
-    status = NormalizeStatus(status);
-    OsConfigLogDebug(log, "SystemCommand: command '%s' returning %d", command, status);
-
-    return status;
-}
-
-#define MAX_COMMAND_RESULT_FILE_NAME 100
-
-int ExecuteCommand(void* context, const char* command, bool replaceEol, bool forJson, unsigned int maxTextResultBytes, unsigned int timeoutSeconds, char** textResult, CommandCallback callback, OsConfigLogHandle log)
-{
-    const char commandTextResultFileTemplate[] = "/tmp/~OSConfig.TextResult%u";
-    const char commandSeparator[] = " > ";
-    const char commandTerminator[] = " 2>&1";
-
-    int status = -1;
-    FILE* resultsFile = NULL;
-    int fileSize = 0;
-    int next = 0;
-    int i = 0;
-    char* commandLine = NULL;
-    size_t commandLineLength = 0;
-    size_t maximumCommandLine = 0;
-    char commandTextResultFile[MAX_COMMAND_RESULT_FILE_NAME] = {0};
-    bool wrappedCommand = false;
-
-    if ((NULL == command) || (0 == system(NULL)))
-    {
-        OsConfigLogDebug(log, "Cannot run command '%s'", command);
-        return -1;
-    }
-
-    commandLineLength = strlen(command);
-    wrappedCommand = (('(' == command[0]) || (')' == command[commandLineLength]));
-
-    // Append a random number to the results file to prevent parallel commands overwriting each other results
-    snprintf(commandTextResultFile, sizeof(commandTextResultFile), commandTextResultFileTemplate, rand());
-
-    commandLineLength += strlen(commandSeparator) + strlen(commandTextResultFile) + strlen(commandTerminator) + (wrappedCommand ? 0 : 2) + 1;
-
-    maximumCommandLine = (size_t)sysconf(_SC_ARG_MAX);
```

Comment:
>_SC_ARG_MAX [](http://example.com/codeflow?start=41&length=11)

Why aren't we any longer needing to watch for __SC_ARG_MAX?

[https://www.man7.org/linux/man-pages/man3/sysconf.3.html](https://www.man7.org/linux/man-pages/man3/sysconf.3.html)

'ARG_MAX - _SC_ARG_MAX
              The maximum length of the arguments to the exec(3) family
              of functions.  Must not be less than _POSIX_ARG_MAX (4096).'

Reply:
execl() will fail with E2BIG, so this check is reduntant.

Reply:
Can we add a test case for this and confirm the E2BIG with it?